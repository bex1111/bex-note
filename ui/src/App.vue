<template>
  <prime-card>
    <template #title>
      <ActionsHeader
      :selectedTitle="selectedTitle"
      :title="title"
      :content="content"
      />
    </template>
    <template #subtitle>
      <TitleSelector v-model="selectedTitle"/>
      <TitleEditor v-model="title" :selected-title="selectedTitle"/>
    </template>
    <template #content>
      <MarkdownEditor v-model="content"></MarkdownEditor>
    </template>
  </prime-card>
  <LoginModal></LoginModal>
  <span class="text-2xl font-medium">Class Warfare</span>
</template>

<script setup>

import MarkdownEditor from "./components/MarkdownEditor.vue";
import ActionsHeader from "./components/ActionsHeader.vue";
import TitleEditor from "./components/TitleEditor.vue";
import LoginModal from "./components/LoginModal.vue";
import TitleSelector from "./components/TitleSelector.vue";
import {ref} from "vue";

const title = ref();
const selectedTitle = ref();
const content = ref();
</script>

<style>
</style>
