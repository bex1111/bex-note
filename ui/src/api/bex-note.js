import axios from 'axios';
import {tokenStore} from '../main';


export const getNoteList = async () => {
    const response = await axios.get('http://localhost:1080/api/internal/note/list', {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const authorize = async (username, password) => {
    const response = await axios.post('http://localhost:1080/api/authorization', {
        username,
        password
    });
    return response.data;
}


export const deleteNote = async (title) => {
    const response = await axios.delete('http://localhost:1080/api/internal/note/delete', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: { title }
    });
    return response.data;
}

export const saveNote = async (title,content) => {
    const response = await axios.delete('http://localhost:1080/api/internal/note/save', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: { title, content }
    });
    return response.data;
}


