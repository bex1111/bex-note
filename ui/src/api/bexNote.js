import axios from 'axios';
import {tokenStore} from '../main';


export const getNoteList = async () => {
    const response = await axios.get('/api/internal/note/list', {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const authorize = async (username, password) => {
    const response = await axios.post('/api/authorize', {
        username,
        password
    });
    return response.data;
}


export const deleteNote = async (title) => {
    const response = await axios.delete('/api/internal/note/delete', {
        headers: {
            'x-auth-token': tokenStore.token
        },
        data: {title}
    });
    return response.data;
}

export const saveNote = async (title, content) => {
    const response = await axios.post('/api/internal/note/save', {
        data: {title, content}
    }, {
        headers: {
            'x-auth-token': tokenStore.token
        }
    });
    return response.data;
}

export const getContent = async (title) => {
    const response = await axios.post('/api/internal/note/content', {
            data: {title}
        },
        {
            headers: {
                'x-auth-token': tokenStore.token
            }
        });
    return response.data;
}
