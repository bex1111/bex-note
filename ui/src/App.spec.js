import {mount} from '@vue/test-utils';
import {vi} from 'vitest';
import App from './App.vue';
import {getNoteList} from './api/bexNote';

vi.mock('./api/bexNote', () => ({
    getNoteList: vi.fn()
}));

describe('App.vue', () => {
    const createWrapper = () => {
        return mount(App, {
            global: {
                stubs: {
                    'prime-card': {
                        template: `<div class="prime-card-stub">
                <div class="card-title"><slot name="title"></slot></div>
                <div class="card-subtitle"><slot name="subtitle"></slot></div>
                <div class="card-content"><slot name="content"></slot></div>
              </div>`
                    },
                    'ActionsHeader': {
                        template: '<div class="actions-header-stub" data-testid="actions-header"></div>',
                        props: ['selectedTitle', 'title', 'content'],
                        emits: ['createNew', 'save', 'delete']
                    },
                    'TitleSelector': {
                        template: '<div class="title-selector-stub" data-testid="title-selector"></div>',
                        props: ['modelValue', 'notes'],
                        emits: ['update:modelValue']
                    },
                    'TitleEditor': {
                        template: '<div class="title-editor-stub" data-testid="title-editor"></div>',
                        props: ['modelValue', 'selectedTitle'],
                        emits: ['update:modelValue']
                    },
                    'MarkdownEditor': {
                        template: '<div class="markdown-editor-stub" data-testid="markdown-editor"></div>',
                        props: ['modelValue', 'title'],
                        emits: ['update:modelValue']
                    },
                    'LoginModal': {
                        template: '<div class="login-modal-stub" data-testid="login-modal"></div>',
                        emits: ['login']
                    }
                }
            }
        });
    }
    const testTitle = 'Test Title';
    beforeEach(() => {
        vi.clearAllMocks();
    });

    it('matches snapshot', () => {
        const wrapper = createWrapper()

        expect(wrapper.html()).toMatchSnapshot();
    });

    describe('createNew', () => {

        it('resets all form values to null when called', async () => {
            const wrapper = createWrapper()

            wrapper.vm.title = testTitle;
            wrapper.vm.selectedTitle = 'Selected Title';
            wrapper.vm.content = 'Test Content';

            expect(wrapper.vm.title).toBe(testTitle);
            expect(wrapper.vm.selectedTitle).toBe('Selected Title');
            expect(wrapper.vm.content).toBe('Test Content');

            await wrapper.vm.createNew();

            expect(wrapper.vm.title).toBeNull();
            expect(wrapper.vm.selectedTitle).toBeNull();
            expect(wrapper.vm.content).toBeNull();
        });
    });

    describe('refreshNotes', () => {
        const mockNotes = [
            {title: 'Note 1'},
            {title: 'Note 2'},
            {title: testTitle}
        ];

        beforeEach(() => {
            getNoteList.mockResolvedValue(mockNotes);
        });

        it('sets selectedTitle to title when title exists in notes', async () => {
            const wrapper = createWrapper();
            wrapper.vm.title = testTitle;
            wrapper.vm.selectedTitle = testTitle + '1';

            await wrapper.vm.refreshNotes();

            expect(wrapper.vm.selectedTitle).toBe(testTitle);
            expect(wrapper.vm.title).toBe(testTitle);
            expect(wrapper.vm.notes).toEqual(mockNotes);
        });

        it('resets selectedTitle and title when delete notes', async () => {
            const wrapper = createWrapper();
            wrapper.vm.title = 'Non-existent Title';
            wrapper.vm.selectedTitle = testTitle;
            wrapper.vm.content = 'Some content';

            await wrapper.vm.refreshNotes();

            expect(wrapper.vm.selectedTitle).toBeNull();
            expect(wrapper.vm.title).toBeNull();
            expect(wrapper.vm.content).toBeNull();
            expect(wrapper.vm.notes).toEqual(mockNotes);
        });

        it('not resets title when save notes the first time', async () => {
            const wrapper = createWrapper();
            wrapper.vm.title = testTitle;
            wrapper.vm.selectedTitle = null;

            await wrapper.vm.refreshNotes();

            expect(wrapper.vm.selectedTitle).toEqual(testTitle);
            expect(wrapper.vm.title).toEqual(testTitle);
            expect(wrapper.vm.notes).toEqual(mockNotes);
        });

    });
});
