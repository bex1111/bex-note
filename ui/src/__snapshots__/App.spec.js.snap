// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > matches snapshot 1`] = `
"<div class=\\"prime-card-stub\\">
  <div class=\\"card-title\\">
    <div class=\\"actions-header-stub\\" data-testid=\\"actions-header\\"></div>
  </div>
  <div class=\\"card-subtitle\\">
    <div class=\\"title-selector-stub\\" data-testid=\\"title-selector\\"></div>
    <div class=\\"title-editor-stub\\" data-testid=\\"title-editor\\"></div>
  </div>
  <div class=\\"card-content\\">
    <div class=\\"markdown-editor-stub\\" data-testid=\\"markdown-editor\\"></div>
  </div>
</div>
<div class=\\"login-modal-stub\\" data-testid=\\"login-modal\\"></div>"
`;
